src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs

```diff
@@ -185,15 +185,15 @@ private async Task<Solution> ProcessResultAsync(CodeFixContext context, Diagnost
                 }
             }
 
-            const SyntaxRemoveOptions options = SyntaxRemoveOptions.KeepUnbalancedDirectives | SyntaxRemoveOptions.AddElasticMarker;
+            var syntaxRemoveOptions = CreateSyntaxRemoveOptions(nodeToRemove);
             if (fieldDocument == propertyDocument)
             {
                 // Same file.  Have to do this in a slightly complicated fashion.
                 var declaratorTreeRoot = await fieldDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
                 var editor = new SyntaxEditor(declaratorTreeRoot, fieldDocument.Project.Solution.Workspace);
-                editor.RemoveNode(nodeToRemove, options);
                 editor.ReplaceNode(property, updatedProperty);
+                editor.RemoveNode(nodeToRemove, syntaxRemoveOptions);
```

Comment:
was moving this important to the fix?

Reply:
Yes, otherwise the test `TestSingleLineWithDirective` won't pass.
The order of change registration matters in this case. If we would delete `nodeToRemove` with directive at first place, the directive remains and will become part of the `property` leading trivia.
The `updatedProperty` doesn't contain this directive and therefore it is lost after the `ReplaceNode` is finished.